src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -668,16 +668,31 @@ public boolean isFips() {
    * @return true if and only if the underlying libcrypto library's FIPS related checks pass
    */
   public boolean isFipsStatusOk() {
-    if (!isFipsSelfTestFailureSkipAbort()) {
+    if (!isFips() || !isFipsSelfTestFailureSkipAbort()) {
       throw new UnsupportedOperationException("ACCP not built with FIPS_SELF_TEST_SKIP_ABORT");
     }
+    if (getSelfTestStatus() == SelfTestStatus.NOT_RUN) {
+      // If FIPS self tests haven't completed, give them a 5s timeout to complete.
+      final long timeout = 5 * 1000;
+      final long deadline = System.currentTimeMillis() + timeout;
+      while (getSelfTestStatus() == SelfTestStatus.NOT_RUN) {
+        try {
+          Thread.sleep(10);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
```

Comment:
Use `com.amazon.corretto.crypto.provider.RuntimeCryptoException` instead of the built-in `RuntimeException`

Reply:
makes sense.